funcr/example_test.go

```diff
@@ -57,6 +58,28 @@ func ExampleUnderlier() {
 	// Output: hello world
 }
 
+func ExamplePseudoStruct() {
+	var log logr.Logger = funcr.NewJSON(
+		func(obj string) { fmt.Println(obj) },
+		funcr.Options{})
+	kv := []interface{}{
+		"field1", 12345,
+		"field2", true,
+	}
+	log.Info("the message", "key", funcr.PseudoStruct(kv))
+	// Output: {"logger":"","level":0,"msg":"the message","key":{"field1":12345,"field2":true}}
+}
+
+func ExampleCallback() {
+	var log logr.Logger = funcr.NewJSON(func(obj string) {
+		fmt.Println(obj)
+	}, funcr.Options{})
+	long := "I have of late but wherefore I know not lost all my mirth"
+
+	log.Info("the message", "key", funcr.Callback(func() interface{} { return strings.Fields(long)[5:9] }))
```

Comment:
Implementation question: does the `funcr.Callback` type actually add any value?

Suppose we just pass `func() interface{}` here and check for that type instead of `funcr.Callback` in the implementation, do we loose any functionality? I don't think so, because logging a function doesn't make sense, so passing pretty much already means "please call me".

The only drawback might be less clarity.

Reply:
I have, in the past, printed pointer values for function pointers, especially in C code where there are no native methods.  My thinking here was that a cast makes it very explicit what the user intends.